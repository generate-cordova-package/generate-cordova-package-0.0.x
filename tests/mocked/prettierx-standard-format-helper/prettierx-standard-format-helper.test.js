/* global jest, it, expect */
/* (see <https://github.com/standard/standard/issues/905>) */

const mockOutputLog = []

const prettierxStandardFormat = require('../../../lib/helpers/prettierx-standard-format')

jest.mock('prettierx', () => ({
  format: (source, options) => {
    // to check that correct options were used *including* parser option
    mockOutputLog.push({ prettierxFormat: { source, options } })

    // (do nothing in this case)
    return source
  }
}))

it('calls prettierx with correct options including correct parser', async () => {
  const testSource = 'console.log(1)'

  const result = prettierxStandardFormat(testSource)

  // quick & simple check
  expect(result.length).not.toBe(0)

  // now ensure correct options including correct parser option
  expect(mockOutputLog).toMatchSnapshot()
})
