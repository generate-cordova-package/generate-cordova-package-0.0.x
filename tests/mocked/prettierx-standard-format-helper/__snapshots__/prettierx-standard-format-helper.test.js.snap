// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls prettierx with correct options including correct parser 1`] = `
Array [
  Object {
    "prettierxFormat": Object {
      "options": Object {
        "parser": "babel",
        "semi": false,
        "singleQuote": true,
        "spaceBeforeFunctionParen": true,
      },
      "source": "console.log(1)",
    },
  },
]
`;
